ArrayAssignmentSet2.js

```diff
@@ -0,0 +1,138 @@
+const movies=[
+    {
+      "MovieName": "The Great Adventure",
+      "ActorName": "John Smith",
+      "ReleaseDate": "2023-01-15"
+    },
+    {
+      "MovieName": "Mystery in the Woods",
+      "ActorName": "Emily Johnson",
+      "ReleaseDate": "2022-09-28"
+    },
+    {
+      "MovieName": "Love and Destiny",
+      "ActorName": "Michael Brown",
+      "ReleaseDate": "2023-05-02"
+    },
+    {
+      "MovieName": "City of Shadows",
+      "ActorName": "Sophia Williams",
+      "ReleaseDate": "2023-03-12"
+    },
+    {
+      "MovieName": "The Last Stand",
+      "ActorName": "William Davis",
+      "ReleaseDate": "2022-11-07"
+    },
+    {
+      "MovieName": "Echoes of Time",
+      "ActorName": "Olivia Wilson",
+      "ReleaseDate": "2022-12-19"
+    }
+];
+
+//1. List the movie name along with the actor name of those movies released in the year 2022
+const movies2022 = movies.filter(movie => new Date(movie.ReleaseDate).getFullYear() === 2022);
+
+movies2022.forEach(movie => {
+    console.log(`${movie.MovieName} - ${movie.ActorName}`);
+});
+
+//2. List the movie names released in the year 2023 where the actor is William Davis.
+const movie2023=movies.filter(movie => new Date(movie.ReleaseDate).getFullYear()===2023 && movie.ActorName==="William Davis");
+movie2023.forEach(movie => {
+    console.log(`${movie.MovieName} - ${movie.ActorName}`);
+});
+
+//3. Retrieve the Actor name and release date of the movie “The Last Stand”
+const LastStand=movies.filter(movie=>movie.MovieName==="The Last Stand");
+LastStand.forEach(movie => {
```

Comment:
return as an object

Reply:
Changes made, Not sure the code is correct